### PR TITLE
chore: update descriptions for config options

### LINKED ISSUE #13239: Use monospaced style for Boolean values, and references to Renovate config options
### Describe the proposed change(s).

https://github.com/renovatebot/renovate/blob/main/lib/config/options/index.ts has `description:` fields, that are also used to generate a part of the config option documentation.

I want to use the monospaced style for things like:

- [x] `true` or `false` Booleans
- [x] References to Renovate concepts like `packageRules`
- [x] References to filenames like `config.js`

Let me know if you want one big PR, or smaller PRs to make this change. 😉

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Update descriptions for config options

## Context

Closes #13239

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -437,7 +437,7 @@ const options: RenovateOptions[] = [
   {
     name: 'optimizeForDisabled',
     description:
-      'Set to `true` to first check for disabling in config before cloning.',
+      'Set to `true` to first check for disabling INSERT WHAT WE CHECK FOR HERE in config before cloning.',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:440]: > Set to `true` to first check for disabling in config before cloning.

This sentence is missing some information about what is disabled in config before cloning.

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -1289,7 +1289,7 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'branchPrefixOld',
-    description: 'Old Prefix to check for existing PRs.',
+    description: 'Old prefix to check for existing PRs.',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:1292]: Just Prefix, seems a bit weird, it should probably be one of these:

- prefix
- `prefix` when referring to an old config option name?

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -2057,7 +2057,7 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'insecureRegistry',
-    description: 'Explicitly turn on insecure Docker registry access (HTTP).',
+    description: 'Explicitly turn on unsafe Docker registry access (HTTP).',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:2060]: I think unsafe is easier to understand than insecure, but feel free to revert this with this suggestion:

```suggestion
    description: 'Explicitly turn on insecure Docker registry access (HTTP).',
```

File: lib/config/options/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/options/index.ts:440]: Is there more info in the markdown file?

Comment (rarkins) [reply lib/config/options/index.ts:440]: I've added markdown docs for this option now

Comment (rarkins) [inline lib/config/options/index.ts:440]:
```suggestion
      'Set to `true` to perform a check for disabled config prior to cloning.',
```

File: lib/config/options/index.ts [hunk 4]
```diff
@@ -1289,7 +1288,7 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'branchPrefixOld',
-    description: 'Old Prefix to check for existing PRs.',
+    description: 'Old prefix to check for existing PRs.',
```
Comment (rarkins) [inline lib/config/options/index.ts:1291]:
```suggestion
    description: 'Old branchPrefix value to check for existing PRs.',
```

File: lib/config/options/index.ts [hunk 5]
```diff
@@ -2057,7 +2056,7 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'insecureRegistry',
-    description: 'Explicitly turn on insecure Docker registry access (HTTP).',
+    description: 'Explicitly turn on unsafe Docker registry access (HTTP).',
```
Comment (rarkins) [inline lib/config/options/index.ts:2059]:
```suggestion
    description: 'Explicitly turn on insecure Docker registry access (HTTP).',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**